src/refract/JSONSchemaVisitor.cc

```diff
@@ -15,6 +15,23 @@
 namespace refract
 {
 
+    template <typename T>
```

Comment:
Hope that it works as expected, as I;m not sure now if in one case we don't want to `clone` even empty elements as is done on line 284 ... but otherwise good move :)